ext/mysqlnd/mysqlnd_enum_n_def.h

```diff
@@ -101,6 +103,10 @@
 #define CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA	(1UL << 21) /* Enable authentication response packet to be larger than 255 bytes. */
 #define CLIENT_CAN_HANDLE_EXPIRED_PASSWORDS		(1UL << 22) /* Don't close the connection for a connection with expired password. */
 #define CLIENT_SESSION_TRACK					(1UL << 23) /* Extended OK */
+/*
+  This is a mysqlnd extension. CLIENT_IGNORE_SIGPIPE is not used anyway. We will reuse it for our case and translate it to forcing the mysql_clear_password protocol
+*/
+#define CLIENT_SEND_CLEAR_PASSWORD		CLIENT_IGNORE_SIGPIPE /* Force plaintext password */
```

Comment:
What exactly is this for? Can we not use `CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA`?

Reply:
The sensible, default behavior of mysqlnd is to use the protocol requested by the server, which is `mysql_native_password`.  This protocol is appropriate for authenticating with username & password.

However, AWS AuroraDB also supports sending an IAM token in place of a password, and this must be sent in the clear.  The problem is that AuroraDB doesn't know which authentication method the client intends to use, so it always sends `mysql_native_password`, and it's up to the client to decide how to send the password data.

Thus, there needs to be a way *force* mysqlnd to send the password in the clear.  That is why I introduced this flag.